sdk/test-utils/recorder/test/browser/recorder.spec.ts

```diff
@@ -0,0 +1,312 @@
+    xhrMock.setup();
+    xhrMock.get("/to/replace", {
+      status: 200,
+      body: expectedHttpResponse
+    });
+
+    // Before starting the recorder, we need to make a copy of the original XHR object, so that we can
+    // restore it before doing the recorder do its magic, in order for the HTTP request to hit the xhr-mock instance.
+    const originalXHR = XMLHttpRequest;
+
+    // The recorder outputs files into the console,
+    // so we need to mock the console.log function to capture and test the recorder output.
+    const originalConsoleLog = console.log;
+    const savedConsoleLogParams: any[] = [];
+    console.log = (...params: any) => {
+      savedConsoleLogParams.push(params);
+    };
+
+    // The recorder should start in the beforeEach call.
+    // We have to do this to emulate that.
+    const fakeThis: TestContextInterface = new TestContext(this.test! as TestContextTest, {
+      ...this.currentTest,
+      file: "test/recorder.browser.spec.ts",
+      // For this test, we don't care what's the content of the recorded function.
+      fn: getNoOpFunction()
+    });
+
+    const recorder = record(fakeThis, recorderEnvSetup);
+
+    // Restoring the XHR, otherwise we can't test this.
+    window.XMLHttpRequest = originalXHR;
+
+    const response = await helloWorldRequest();
+
+    // This test's request reached the server and received the expected response.
+    expect(response).to.equal(expectedHttpResponse);
+
+    // Cleaning everything before we continue verifying the results.
+    xhrMock.teardown();
+    recorder.stop();
+    console.log = originalConsoleLog;
+
+    // Here we confirm that the recorder generated an expected output on the console.logs.
+    // This output is used to generate the recording files in the filesystem, though here we're only
+    // checking what was that the recorded emitted to the standard output.
+    expect(savedConsoleLogParams[0][0]).to.equal(
+      // The recordings here are empty because we hijacked the XHR.
+      // See the playback test for an example of a properly constructed recording object.
+      // TODO: Find a way to capture the complete output.
+      JSON.stringify({
+        writeFile: true,
+        path:
+          "./recordings/browsers/the_recorders_public_api_on_a_browser/recording_should_record_a_simple_test.json",
+        content: {
+          recordings: [],
+          uniqueTestInfo: {
+            uniqueName: {},
+            newDate: {}
+          },
+          hash: MD5(getNoOpFunction().toString())
+        }
+      })
+    );
+  });
+
+  it("should playback a simple test", async function() {
+    // Setting up the playback mode.
+    // The PATH environment variable is not needed on playback.
+    // The recorder will assume that it is in playback mode by default.
+    // (window as any).__env__.TEST_MODE = "playback";
+
+    // This is to emulate what 'karma-json-preprocessor' does for us during the real scenarios.
+    (window as any).__json__ = {
```

Comment:
since we do this pattern a lot, maybe a small helper that is like 

```ts
function setOnWindow(name, value) {
   (window as any)[name] = value;
}
```

would be helpful?

Reply:
I am not finding this particularly useful. Do you mind sharing more of your line of thinking? I want to be able to appreciate your point of view.

Reply:
Say we want to reduce/eliminate the usage of `any`, it's easier to ignore/whitelist one helper than it is to get comfortable seeing it everywhere in code.

The more we duplicate a pattern, the more likely someone who is new to the codebase will see it and use it.

I don't feel strongly about this one.

Reply:
I like this! I'll take it.

Reply:
Here's my approach: https://github.com/Azure/azure-sdk-for-js/pull/7254/commits/d1a82cb4d9877686748ca41aeb123f4a300b79ed#diff-c4fb374303df7c0ca09e2a6633dd4673R475

I took your idea and I made it loosely speaking a lens, with a getter and a setter, and a very simple path traversal mechanism. I switched all reads and writes to the window object to use windowLens, and I added two unit tests to represent setting and getting at a single level, and at three levels deep. Let me know what you think!

Reply:
(I did a related commit directly afterwards, but it didn't change the implementation, only some use cases)